mos_tests/conftest.py

```diff
@@ -167,19 +167,31 @@ def set_openstack_environ(fuel_master_ip):
             os.environ[k] = v
 
 
-@pytest.fixture
-def os_conn(env):
-    """Openstack common actions"""
-    os_conn = OpenStackActions(
-        controller_ip=env.get_primary_controller_ip(),
-        cert=env.certificate, env=env)
+def get_os_conn(environment):
+    os_connection = OpenStackActions(
+        controller_ip=environment.get_primary_controller_ip(),
+        cert=environment.certificate, env=environment)
 
-    wait(os_conn.is_nova_ready,
+    wait(os_connection.is_nova_ready,
          timeout_seconds=60 * 5,
          expected_exceptions=Exception,
          waiting_for="OpenStack nova computes is ready")
     logger.info("OpenStack is ready")
-    return os_conn
+    return os_connection
+
+
+@pytest.fixture(scope='class')
```

Comment:
In this case you need to change a bit cleanup fixture logic (not revert env if test is undestructive).